src/utils/source.js

```diff
@@ -199,6 +199,10 @@ function getMode(source: Source) {
   return { name: "text" };
 }
 
+function isLoaded(source: Source) {
+  return source.loadedState === "loaded";
```

Comment:
i noticed that we are doing toJS() a lot to get the object for this function. An improvement here would be to do:
```
return source.get("loadedState") === "loaded"
```

toJS() is an expensive function and we should avoid it where we can. I noticed there is one case where we have a regular source object, in findSourceMatch. it might be that in that case we are also doing .toJS() -- i will take a look